BlazarUI/app/scripts/components/shared/BuildingIcon.jsx

```diff
@@ -3,7 +3,13 @@ import React, {Component, PropTypes} from 'react';
 class BuildingIcon extends Component {
 
   getClassNames() {
-    return `building-icon sidebar__active-building-icon building-icon--${this.props.result} ${this.props.classNames} building-icon--${this.props.size}`;
+    let prevBuildStateModifier = ``;
+
+    if (this.props.result === "IN_PROGRESS" && this.props.prevBuildState) {
```

Comment:
> Though this will look a little different once using `classnames`.

What did you have in mind? I've updated the function to use `classnames`, but the structure of the function is still similar